### ISSUE TITLE
Add custom headers support to SES sendBulkTemplatedEmail

### ISSUE THREAD

**Issue by chrisc0** [aws/aws-sdk-php#2836]
### Describe the feature

I need to be able to add custom email headers while sending out hundreds (potentially thousands) of emails at a time using the SES sendBulkTemplatedEmail command but it doesn't currently seem possible.

### Use Case

I need to be able to add custom email headers while sending out hundreds (potentially thousands) of emails at a time using the SES sendBulkTemplatedEmail command but it doesn't currently seem possible.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### SDK version used

latest

### Environment details (Version of PHP (`php -v`)? OS name and version, etc.)

php 8.1.2, Ubuntu 22.04

**Comment by nicodemuz** [comment]
Below quotes are from the recent [AWS article](https://aws.amazon.com/blogs/messaging-and-targeting/an-overview-of-bulk-sender-changes-at-yahoo-gmail/):

> "In a move to safeguard user inboxes, [Gmail](https://blog.google/products/gmail/gmail-security-authentication-spam-protection/) and [Yahoo Mail](https://blog.postmaster.yahooinc.com/post/730172167494483968/more-secure-less-spam) announced a new set of requirements for senders effective from February 2024."

> 2. Set up an easy unsubscribe for email recipients
> Bulk senders are expected to include a mechanism to unsubscribe by adding an easy to find link within the message. The February 2024 mailbox provider rules will require senders to additionally add one-click unsubscribe headers as defined by [RFC 2369](https://datatracker.ietf.org/doc/html/rfc2369) and [RFC 8058](https://datatracker.ietf.org/doc/html/rfc8058). These headers make it easier for recipients to unsubscribe, which reduces the rate at which recipients will complain by marking messages as spam.

> To set up one-click unsubscribe without using the SES subscription management feature, include both of these headers in outgoing messages:
> 
> List-Unsubscribe-Post: List-Unsubscribe=One-Click
> List-Unsubscribe: <https://example.com/unsubscribe/example>

How can we use [sendBulkTemplatedEmail](https://docs.aws.amazon.com/ses/latest/APIReference/API_SendBulkTemplatedEmail.html) API to add List-Unsubscribe-Post and List-Unsubscribe headers? With a quick look at the API documentation, I don't see any ways to add these headers. We do not wish to use SES subscription management feature as we already have our subscription management page where the user can unsubscribe.

Going forward, this seems like an essential feature that needs to be supported.

**Comment by DevKishan** [comment]
> Going forward, this seems like an essential feature that needs to be supported.
Yes, this is indeed a necessary feature when sending emails to subscribers in bulk. Without this, we are forced to send emails one subscriber at a time it is not possible to add the **List-Unsubscribe** custom header through **API_SendBulkTemplatedEmail** API. This point in [this](https://docs.aws.amazon.com/ses/latest/dg/sending-email-subscription-management.html#:~:text=The%20List%2DUnsubscribe%20header%20and%20ListManagementOptions%20footer%20links%20are%20added%20only%20if%20the%20email%20is%20being%20sent%20to%20a%20single%20recipient.) SES doc further confirms it.

![image](https://github.com/aws/aws-sdk-php/assets/21952931/46566784-9ff6-4ed1-86a2-99b577e34ee2)


Other competitor ESPs like Sendgrid, Mailgun do provide the option to add List-Unsubscribe and List-Unsubscribe-Post headers in bulk mode as well.

**Comment by HashimHaider** [comment]
Any update on this? Given the recent bulk sender changes at Yahoo/Gmail (dated February, 2024), we need the List-Unsubscribe header more than ever. 
 https://aws.amazon.com/blogs/messaging-and-targeting/an-overview-of-bulk-sender-changes-at-yahoo-gmail/

**Comment by gileri** [comment]
In a support ticket related to this bulk sender policy changes, the AWS representative said that something was cooking internally in order to support adding `List-Unsubscribe` headers in bulk-sending SES API calls. No specifics or planned release date as usual.

**Comment by serbanveres** [comment]
+1

**Comment by TomWaterz** [comment]
Any update on this?
